tests/common/plugins/custom_markers/__init__.py

```diff
@@ -43,13 +43,19 @@ def pytest_runtest_setup(item):
         check_device_type(item)
 
 def check_topology(item):
-    toponames = [mark.args for mark in item.iter_markers(name="topology")]
+    # The closest marker is used here so that the module or class level
+    # marker will be overrided by case level marker
+    toponames = item.get_closest_marker("topology")
     if toponames:
-        cfg_topos = item.config.getoption("--topology").split(',')
-        if all(topo not in toponames[0] for topo in cfg_topos):
-            pytest.skip("test requires topology in {!r}".format(toponames))
+        if "any" in toponames.args:
+            # match any topology
+            pass
+        else:
```

Comment:
This check is not needed. we don't want to run testcases marked with "any" when we specify any topology (say t0) in the command line because that will mix up our results. That was the reason we came up the the marker "any".

Reply:
Thanks Neetha. I've made the update. Please review again.